common/src/main/java/com/wynntils/core/net/ApiResponse.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.net;
+
+import java.io.InputStream;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class ApiResponse extends NetResult {
+    private static final Map<HttpRequest, CompletableFuture<InputStream>> DOWNLOAD_FUTURES = new HashMap<>();
```

Comment:
No, it will not be used in #826 either. Qodana is right but wrong. :-) It is not read, that is true, but storing the Future in a static map like this means the Future object will be kept alive not reclaimed by the GC.

It is possible that this is not necessary, and that the Future has some mechanism to keep the instance alive as long as it is not completed, but I am not sure, and I could find no documentation on this. 

If any of you know for certain how this works, I can remove this. But oterhwise, I'd like to keep it for now. If the Future could be GC'ed before all the async processing has happened, then we might get intermittent failures that can be very tricky to track down -- the likelihood that a GC will happen just after creating the Future but before calling the async handling part is very small.

So what I do is that I store the Future in a map, when it is created, and then ask the future to remove itself as it's last operation.

I see now that it could have been made a List or Set instead of Map. I had it a Set originally, but then I actually needed to look up things and made a Map out of it. Then I could get rid of that need, but forgot to change it back to a Set.

Reply:
I am fine with it, just make a tldr comment about this. Even if it is supposed to work, maybe it would be dependant on jvm version/vendor, so it is safer this way